### PR TITLE
EL-4277 - Combobox: Issue with combox controls and custom validators

### PR DESCRIPTION
**Pull request by JonathanToddMF**
#### Checklist

* [x] The contents of this PR are mine to submit. No third party code or other material is being committed.
* [x] New third party dependencies (if any) are linked via `package.json`. Third party dependencies are licensed under one of the following: Apache, MIT, BSD, Mozilla Public License, Eclipse Public License, or Oracle Binary Code License.
* [x] The code in this PR does not contain export-restricted encryption algorithms.

#### Ticket / Issue

https://portal.digitalsafe.net/browse/EL-4277

#### Description of Proposed Changes

Added example of how to implement validation
Additional formatting changes similar to that of the previous documentation changes

#### Documentation CI URL

https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-4277-validation-docs

#### Downstream Build(s)

* [caf/ux-aspects-micro-focus](https://jenkins.swinfra.net/job/SEPG/job/caf/job/caf~ux-aspects-micro-focus~EL-4277-validation-docs~CI/)

### CODE REVIEW

**Comment by JonathanToddMF** [comment]
Old PR link - https://github.com/UXAspects/UXAspects/pull/1596

File: docs/app/pages/components/components-sections/select/select/snippets/validation.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	get select(): AbstractControl {
+		return this.formGroup.get('select');
+	}
+
+	get checkbox(): AbstractControl {
+		return this.formGroup.get('checkbox');
+	}
+
+	ngOnInit() {
```
Comment (ashley-hunter) [inline docs/app/pages/components/components-sections/select/select/snippets/validation.ts:29]:
```suggestion
	ngOnInit(): void {
```

File: docs/app/pages/components/components-sections/select/select/snippets/validation.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+export class AppComponent implements OnInit, OnDestroy {
+
+	formGroup: FormGroup = new FormGroup({
+		checkbox: new FormControl(false, [Validators.required]),
+		select: new FormControl({ value: undefined, disabled: true }, [Validators.required]),
+	});
+
+	options: string[] = ['Option1', 'Option2'];
+
+	private _onDestroy = new Subject<void>();
```
Comment (ashley-hunter) [inline docs/app/pages/components/components-sections/select/select/snippets/validation.ts:19]:
```suggestion
	private _onDestroy$ = new Subject<void>();
```

**Review by ashley-hunter — CHANGES_REQUESTED** [review]
Thanks, looks good, just had two minor suggestions for consistency.

File: docs/app/pages/components/components-sections/select/select/snippets/validation.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { OnDestroy, OnInit } from '@angular/core';
+import {
+	AbstractControl, FormControl,
+	FormGroup,
+	Validators
+} from '@angular/forms';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+export class AppComponent implements OnInit, OnDestroy {
```
Comment (ashley-hunter) [inline docs/app/pages/components/components-sections/select/select/snippets/validation.ts:10]:
```suggestion
import { OnDestroy, OnInit, Component } from '@angular/core';
import {
	AbstractControl, 
	FormControl,
	FormGroup,
	Validators
} from '@angular/forms';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';

@Component({
    selector: 'app',
    templateUrl: './app.component.html'
})
export class AppComponent implements OnInit, OnDestroy {
```

**Review by ashley-hunter — CHANGES_REQUESTED** [review]
Thanks, looks good and seems to work well, I just had one final suggestion to include the Component decorator just so someone can easily copy and paste the example code into plunker or a file without having to modify it. It caught me out when I went to check it in plunker so its possible others might have the same issue. Apart from that, all good.

**Review by ashley-hunter — DISMISSED** [review]
All good, thanks

**Comment by JonathanToddMF** [comment]
Just adding you @donna-marijne for consistency with the inner source pr

File: docs/app/pages/components/components-sections/select/select/select.component.html [hunk 4]
```diff
@@ -311,6 +314,29 @@
     </tr>
 </uxd-api-properties>
 
+<h4>Validation</h4>
+<p>
+    The <code>ux-select</code> component can have validation applied to it like any
+    other angular form element.
```
Comment (donna-marijne) [inline docs/app/pages/components/components-sections/select/select/select.component.html:320]:
```suggestion
    other Angular form element.
```

File: docs/app/pages/components/components-sections/select/select/select.component.html [hunk 5]
```diff
@@ -311,6 +314,29 @@
...
+</p>
+<p>
+    Do not use the <code>disabled</code> or <code>required</code> inputs when using Reactive Forms. Use the Reactive
+    Forms API to achieve this instead.
```
Comment (donna-marijne) [inline docs/app/pages/components/components-sections/select/select/select.component.html:324]: Link to the documentation of the API mentioned.

File: docs/app/pages/components/components-sections/select/select/snippets/validation.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<form [formGroup]="formGroup">
+    <div class="form-group">
+        <ux-checkbox formControlName="checkbox">Activate select</ux-checkbox>
+    </div>
```
Comment (donna-marijne) [inline docs/app/pages/components/components-sections/select/select/snippets/validation.html:4]: Remove the checkbox to simplify the example, the intent is to show how to apply validation to the select component only.

**Review by donna-marijne — CHANGES_REQUESTED** [review]

**Review by donna-marijne — APPROVED** [review]

**Pull request merged**